File: 2706-buy-two-chocolates/2706-buy-two-chocolates.py
class Solution:
    def buyChoco(self, prices: List[int], money: int) -> int:
        f, s = float(inf), float(inf)
        for x in prices:
            if x < f:
                s = f
                f = x
            else: s = min(s, x)
        return money-s-f if s+f <= money else money

File: 2706-buy-two-chocolates/2706-buy-two-chocolates.cpp
class Solution {
public:
    int buyChoco(vector<int>& prices, int money) {
        int small = INT_MAX, secSmall = INT_MAX;
        for(int x : prices){
            if(x < small){
                secSmall = small;
                small = x;
            }
            else secSmall = min(secSmall, x);
        }
        return small+secSmall<=money ? money-small-secSmall : money;
    }
};